### PR TITLE
Feat: Create food-adding drawer (UI)

### PR DESCRIPTION
**Pull request by uskayyyyy**
## Overview

This PR creates the food-adding button [ + ] and drawer and applies the design.

## Changes

- Adjusted the mobile layout paddings.
- Indexed all the feature-specific components
- Added the ring style to the SelectTrigger button when the select popover is open

## Review points

- When you review, please focus on the UI and components structure.
- May refactor when creating a food-editing drawer to share similar components.

## Screen Captures

Added the [ + ] button and the drawer.

https://github.com/genesis-tech-tribe/nishiki-frontend/assets/99148565/5d2a5bcf-d311-4a0a-9138-61fd96c07132

## Note

-

## Assignee Checklist:

- [x] The base branch is correct (See: [Types of Branches](https://github.com/genesis-tech-tribe/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#types-of-branches))
- [x] The branch name follows the [Branch Naming Conventions](https://github.com/genesis-tech-tribe/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#branch-naming-conventions)
- [x] The correct assignees and reviewers have been designated for this PR
- [x] The coding style follows the [Coding Style Guide](https://github.com/genesis-tech-tribe/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [x] All the related issues are associated with this PR
- [x] All criteria in the associated issues are met (please tick the checkboxes)
- [x] My changes do not generate new warnings or errors (especially in the console)

## Reviewer Checklist:

- [ ] The code follows the generic best practices and our [Coding Style Guide](https://github.com/genesis-tech-tribe/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [ ] Code is well-commented and easy to understand
- [ ] UI changes accurately reflect the provided design

### CODE REVIEW

File: src/features/foods/components/FoodFilterCategoryDrawer.tsx
Comment (uskayyyyy) [inline src/features/foods/components/FoodFilterCategoryDrawer.tsx:1]: In this file, I only removed the unnecessary root `<div>` and replaced `<button>` tags with `<Button>`

File: src/features/foods/components/AddEditForm/FormDatePicker.tsx
Comment (uskayyyyy) [inline src/features/foods/components/AddEditForm/FormDatePicker.tsx:1]: This component is a DatePicker component, especially for the Add/Edit food form, which expects to receive `<input type="hidden">` as a prop.

File: src/features/foods/components/AddEditForm/CategorySelect.tsx
Comment (uskayyyyy) [inline src/features/foods/components/AddEditForm/CategorySelect.tsx:1]: After consideratioin, I decided not to share the category-select component with the food filter.

File: src/features/foods/components/AddEditForm/AddDrawerBody.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+          <Input name="quantity" id="quantity" type="number" />
+        </LabeledInput>
+        <LabeledInput label="Unit" htmlFor="unit">
+          <Input name="unit" id="unit" type="text" />
+        </LabeledInput>
+      </div>
+      <LabeledInput label="Expiry" htmlFor="expiry">
+        <FormDatePicker
+          id="expiry"
+          hiddenInput={(date) => <Input type="hidden" name="expiry" value={date} />}
```
Comment (uskayyyyy) [inline src/features/foods/components/AddEditForm/AddDrawerBody.tsx:120]: This component applies the technique called [render prop(s)](https://react.dev/reference/react/cloneElement#passing-data-with-a-render-prop).

I implemented this in order to define a state and component together in the child component while making the state accessible from the parent component.

However, this approach seems to be getting less common. Thus, I may replace it with the HOC or custom hook shortly.

File: src/components/page/FoodsPage.tsx [hunk 2]
```diff
@@ -2,11 +2,14 @@
 
 import { MenuMeatballIcon } from '@/assets/images/icons';
 import { Icon } from '@/components/ui';
-import { BadgeList } from '@/features/foods/components/BadgeList';
-import { FilterButton } from '@/features/foods/components/FilterButton';
-import { FoodList } from '@/features/foods/components/FoodList';
-import { FoodSort } from '@/features/foods/components/FoodSort';
-import { SearchBar } from '@/features/foods/components/SearchBar';
+import {
```
Comment (kotaaaa) [inline src/components/page/FoodsPage.tsx:5]: Thank you for fixing these 🙇‍♂

**Review by kanta1207 — COMMENTED** [review]
### [Comment / Question]
You named directory name `AddEditForm`, so this os  gonna be the directory to contain `EditButton` and `EditDrawerBody`, `EditDrawerContent`?

If my understanding is correct. I like the approach, but at the same time, I feel like separating `AddForm` and `EditForm` directory could be also an option. In that case, we're gonna also create new directory to put all the add/edit form related components?

**Review by kotaaaa — APPROVED** [review]
All looks good! Thank you for your work!!:)

**Comment by uskayyyyy** [comment]
@kotaaaa 
Thank you for your review and approval!

@kanta1207 
> You named directory name AddEditForm, so this os gonna be the directory to contain EditButton and EditDrawerBody, EditDrawerContent? If my understanding is correct. I like the approach, but at the same time, I feel like separating AddForm and EditForm directory could be also an option. In that case, we're gonna also create new directory to put all the add/edit form related components?

I understand your point. I will rearrange the directory structure that you pointed out in the next PR of implementing the food-editing modal because I have already commenced it.

**Pull request merged**